hazelcast/src/main/java/com/hazelcast/map/impl/AbstractEvictableRecordStore.java

```diff
@@ -382,6 +383,16 @@ protected void accessRecord(Record record, long now) {
         setExpirationTime(record, maxIdleMillis);
     }
 
+    protected void mergeRecordExpiration(Record record, EntryView mergingEntry) {
+        if (mergingEntry.getTtl() > 0 && mergingEntry.getTtl() != Long.MAX_VALUE) {
+            record.setTtl(mergingEntry.getTtl());
+        }
+        if (mergingEntry.getExpirationTime() > 0 && mergingEntry.getExpirationTime() != Long.MAX_VALUE) {
+            record.getStatistics().setExpirationTime(mergingEntry.getExpirationTime());
```

Comment:
Goodness, yes massive NPE problem; thank you!

I was thinking that ```ExpirationTimeSetter#setExpirationTime``` would not be appropriate because it regenerates the expiration time from the entries TTL at the point of WAN replication receipt, rather than synchronising the appropriate value that was set at the point of remote creation.

However this raises an interesting dilemma in terms of if the ```max-idle-seconds``` is different between remote clusters this wouldn't be honoured as the calculation of the expiration time was based on the configuration of source cluster... perhaps unlikely, but feasible.

Basically either I fix up the NPE issue, as such...
```
        if (record.getStatistics() != null) {
            record.getStatistics().setExpirationTime(mergingEntry.getExpirationTime());
        }
```

or use ```ExpirationTimeSetter#updateExpiryTime``` to both set the TTL and expiration time (admittedly having to source the appropriate ```MapConfig``` in order to get ```max-idle-seconds```), but the entry will effectively have a later expiry that the source cluster.

Thoughts? I prefer the former as it is less vulnerable to any async WAN replication delays.